tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -1611,12 +1611,10 @@ bool DeclResultIdMapper::decorateResourceBindings() {
 
     for (uint32_t i = 0; i < numBindingsToUse; ++i) {
       bool success = bindingSet.useBinding(bindingNo + i, setNo);
-      if (!success && spirvOptions.flattenResourceArrays) {
```

Comment:
I think it would be better to let the user know "binding numbers are overlap" using **warning** and explain "it is not a good practice but spirv-opt will solve this. If it bothers you, you can just simple keep it"?

I do believe that some users meet this situation because of mistakes. An explicit explanation may be helpful for some users.

Reply:
the problem is that the warning can be invalid if the optimizer fully eliminates a variable. In all other cases we have also put the responsibility of correct descriptor set and binding number of the user. Similarly, we take user's input about offsets (e.g. `:register(...)` or `[[vk::offset(...)]]` and use it, putting the responsibility of correctness on the user.

Reply:
Hmm.. then we may give the warning only when `-fcgl` is given, but probably the legalization will fix it, right?

Reply:
it's possible.